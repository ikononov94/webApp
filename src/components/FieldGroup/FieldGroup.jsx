import React from 'react';
import { FormGroup, FormControl, ControlLabel } from 'react-bootstrap';

const FieldGroup = ({ label, ...props }) => (
  <FormGroup>
    <ControlLabel>{label}</ControlLabel>
    <FormControl {...props} />
  </FormGroup>
);

export default FieldGroup;
