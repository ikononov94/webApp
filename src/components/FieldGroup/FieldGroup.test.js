import React from 'react';
import { shallow } from 'enzyme';
import { ControlLabel, FormControl } from 'react-bootstrap';
import FieldGroup from './FieldGroup';

describe('Component <FieldGroup />', () => {
  const wrapper = shallow(<FieldGroup label="label" type="text" value="value" />);

  it('render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ControlLabel', () => {
    expect(wrapper.find(ControlLabel).length === 1).toEqual(true);
  });

  it('should render FormControl', () => {
    expect(wrapper.find(FormControl).length === 1).toEqual(true);
  });
});
