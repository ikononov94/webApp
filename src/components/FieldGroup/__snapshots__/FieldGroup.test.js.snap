// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FieldGroup /> render 1`] = `
<FormGroup
  bsClass="form-group"
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    label
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="input"
    type="text"
    value="value"
  />
</FormGroup>
`;
